tests/cli/test_install_tar.sh

```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+# Note: execute this file from the project root directory
+
+#####
+#
+# Builds tar image and install it using liveimg kickstart command
+#
+#####
+
+set -e
+
+. /usr/share/beakerlib/beakerlib.sh
+. $(dirname $0)/lib/lib.sh
+
+CLI="${CLI:-./src/bin/composer-cli}"
+QEMU_BIN="/usr/bin/qemu-system-$(uname -m)"
+QEMU="$QEMU_BIN -machine accel=kvm:tcg"
+
+
+rlJournalStart
+    rlPhaseStartSetup
+        if ! rlCheckRpm httpd; then
+            dnf -y install httpd
+        fi
+        systemctl start httpd
+
+        ks_path="/var/www/html/ks-tar.cfg"
+        cat > $ks_path << __EOF__
+lang en_US.UTF-8
+timezone America/New_York
+keyboard us
+rootpw redhat
+bootloader --location=mbr
+zerombr
+clearpart --initlabel --all
+autopart
+
+liveimg --url http://10.0.2.2/root.tar.xz
+__EOF__
+
+        tmp_dir=(mktemp -d /tmp/composer.XXXXX)
+        ssh_key_dir=$(mktemp -d /tmp/composer-ssh-keys.XXXXXX)
+
+        rlRun -t -c "ssh-keygen -t rsa -N '' -f $ssh_key_dir/id_rsa"
+        pub_key=(cat $ssh_key_dir/id_rsa.pub)
+
+        bp_name="test-tar"
+        blueprint="$bp_name.toml"
+        cat > $blueprint << __EOF__
+name = "$bp_name"
+description = "tar image test"
+version = "0.0.1"
+modules = []
+
+# temporary
+[[packages]]
+name = "grubby-deprecated"
+version = "*"
+
+[[packages]]
+name = "kernel"
+version = "*"
+
+[[packages]]
+name = "openssh-server"
+version = "7.*"
+
+[[groups]]
+name = "anaconda-tools"
+
+[[customizations.user]]
+name = "root"
+key = "$pub_key"
+__EOF__
+        composer-cli blueprints push $blueprint
+        image_path="/var/www/html/root.tar.xz"
+
+        version=$(awk -F = '$1 == "VERSION_ID" { print $2 }' /etc/os-release)
+        arch=$(uname -m)
+        baseurl=$(curl "https://mirrors.fedoraproject.org/mirrorlist?repo=fedora-${version}&arch=${arch}" | \
+            grep -v "^#" | head -n 1)
+        curl --remote-name-all $baseurl/images/pxeboot/{vmlinuz,initrd.img}
```

Comment:
@bcl I am wondering if lorax-composer keeps a reference (or has an easy access to that reference) to the kernel/initrd used during building the image (i.e. from anaconda)?

If we could grab these from a local directory referenced by the compose UUID it would be much more easier instead of trying to figure out what distro we are running on and where to get these files from.

Reply:
If the blueprint includes the kernel the final-kickstart.ks will have the version in it.